### PR TITLE
Updates with new builds

### PR DESCRIPTION
**Pull request by DRSDavidSoft**
### description
This the `rdpwrap.ini` that I personally use, with cleaned up syntax and missing builds from your one

### added missing builds
- `[10.0.18908.1000]`

**edit:** other proposed missing builds were apparently added in 9abffc4

#### `[10.0.17763.165]`
|  alt | current |
| --- | --- |
| SingleUserOffset.x64=132F9 | SingleUserOffset.x64=1322C |

**edit:** `1322C` seems to be correct.
#### `[10.0.17763.437]`
|  **correct**| current |
| --- | --- |
| SingleUserOffset.x64=1322C | SingleUserOffset.x64=1339C |

**edit:** the value in your branch has been fixed via d96866a
- source: https://github.com/stascorp/rdpwrap/pull/725/files#r292265309
- confirmation: https://github.com/stascorp/rdpwrap/issues/729#issuecomment-482609476
#### `[10.0.18362.1]`
|  alt | current |
| --- | --- |
| SingleUserOffset.x64=0DDC9 | SingleUserOffset.x64=0DBFC |
Not sure which one is correct

### CODE REVIEW

**Comment by asmtron** [comment]
Hi DRSDavidSoft,

thanks for your participation. The following SingleUserOffset values are correct:

```
[10.0.17763.165] the value of SingleUserOffset.x64=1322C is correct
.text:0000000180013229 C7 40 08 01 00 00 00                          mov     dword ptr [rax+8], 1

[10.0.17763.437] the value of SingleUserOffset.x64=1339C is correct
.text:0000000180013399 C7 40 08 01 00 00 00                          mov     dword ptr [rax+8], 1

[10.0.18362.1] the value of SingleUserOffset.x64=0DBFC is correct
.text:000000018000DBF9 C7 40 08 01 00 00 00                          mov     dword ptr [rax+8], 1
```

**Comment by DRSDavidSoft** [comment]
@asmtron Thank you for the correct values. It seems (for at least) `10.0.17763.437`, the [alternative one](https://github.com/stascorp/rdpwrap/pull/725/files#r292265309) also works.

In this branch, I have kept the values that you suggest are correct.

At this point, this PR only adds `10.0.18908.1000` in addition to some general clean up (see [here](https://github.com/asmtron/rdpwrap/pull/1/files)).

Would you please merge it so I can rebase my branch off of yours as upstream? :)
Thank you for maintaining it!

File: res/rdpwrap.ini [hunk 1]
```diff
@@ -3997,7 +4017,6 @@ bServerSku.x86        =C3F74
 ulMaxDebugSessions.x86=C3F78
 bRemoteConnAllowed.x86=C3F7C
 
-zlMaxUserSessions.x64  =F23B0
```
Comment (DRSDavidSoft) [inline res/rdpwrap.ini:4000]: @asmtron `zlMaxUserSessions` is bogus and therefore should be removed

**Comment by asmtron** [comment]
new ini updates 2020

**Comment by graphixillusion** [comment]
@asmtron Hi there. Now that the main repo is closed, what about to open issues section over here? Btw, thank you for the last update. Do you have fix for version 2004 too?

**Comment by shippj** [comment]
thank you so much for the quick update @asmtron .  Since the main repo was closed "due to a violation of GitHub's terms of service", would you mind picking a new meeting place and posting it here just in case your repo gets closed too?  Maybe a decentralized system, such as mastodon?

**Comment by shippj** [comment]
win10 32bit is still showing the "Another user is signed in. If you continue, they'll be disconnected" message.

image

**Comment by asmtron** [comment]
PLease send me the termsrv.dll of your win10 32bit (C:\Windows\System32\termsrv.dll) [version 10.0.18362.836]
 

Sent: Saturday, May 16, 2020 at 1:09 AM

**Comment by shippj** [comment]
here ya go, thanks!

[termsrv.zip](https://github.com/asmtron/rdpwrap/files/4640792/termsrv.zip)

**Comment by shippj** [comment]
so, i'm trying to learn how to do this myself.  I used to play around with IDA back about 15-20 years ago but haven't had the need to mess with it since and i've forgotten most of what I knew.  I was delighted to see that IDA is now free for home use, and it works great on Linux, so I fired it up and found a "find THIS hex string and replace it with THIS hex string" tutorial for cracking the 18362.657 version of the DLL.  After learning how to find that hex string in IDA, I noticed the address was the same as the DefPolicyOffset in the INI.  So, I looked at the 32bit version of the 18362.836 DLL in IDA and found offset 50269 (without cheating!).  I looked for the "cdefpolicy query failed license not available" string.  Then, I confirmed in the INI that was the same DefPolicyOffset as the 18362.657 DLL.  So I feel like I understand what DefPolicyOffset is, but I don't understand what the others are and I probably wouldn't be able to find them in a different version of the DLL.

LocalOnlyOffset
image

SingleUserOffset (I might could find this one)
image

SLInitOffset (some kind of api import or export?)
image

So what do you think?  Am I trainable to be elite like you where I could maybe help you guys maintain the INI, or should I just accept the fact that i'm a lamer and always will be and I just need to leave the real stuff up to you guys?

**Comment by asmtron** [comment]
Hi,

 

thx,

 


add support for termsrv.dll 10.0.18362.836 (x86), you will find the right function with address much easier if you compare the new termsrv.dll with an older working termsrv.dll


 


Sent: Sunday, May 17, 2020 at 6:44 PM

**Pull request closed without merging**